examples/flaskr/flaskr/__init__.py

```diff
@@ -4,10 +4,16 @@
 from flask import Flask
 from flask.cli import with_appcontext
 from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy.orm import DeclarativeBase
 
 __version__ = (1, 0, 0, "dev")
 
-db = SQLAlchemy()
+
+class Base(DeclarativeBase):
```

Comment:
I don't think this is needed at all. `db = SQLAlchemy()` works fine even for SA 2, no?

If this is actually needed to use the `Mapped` notation this feels like something that needs to be fixed in FSA - because having to do this here is super ugly IMHO.

Reply:
I'm not sure if you saw the discussion in https://github.com/pallets-eco/flask-sqlalchemy/issues/1140 but yes, the design we settled on does require you to pass in a base class. This gives you the flexibility to use MappedAsDataclass, DeclarativeBaseNoMeta, etc, instead of us deciding on what's in your base class.

Reply:
nope, hadn't seen it or at least I don't remember it... feels a bit obscure when you just have a single superclass but makes more sense if you use multiple superclasses/mixins

Reply:
We'd like to make the 2.0 style the default at some point, but it will require a bit more thought, and we didn't want to block being able to use it at all in this next release.